okhttp/src/jvmMain/kotlin/okhttp3/internal/connection/RealConnection.kt

```diff
@@ -90,6 +92,15 @@ class RealConnection(
    * Once true this is always true. Guarded by this.
    */
   var noNewExchanges = false
+    set(value) {
+      val previousValue = field
+      field = value
+      if (!previousValue) {
+        queueEvent {
+          connectionListener.noNewExchanges(this)
```

Comment:
This is the one that feels unfortunate... Relies on `connection.synchronizedWithEvents`

Reply:
Yeah I think we should just pull up the event-notifying side-effect into the callsite